### PR TITLE
Fixing keycloak setup

### PR DESCRIPTION
**Pull request by Fosol**
## Description

Fixed keycloak initialization.
Cleaned up `make` file.
Tested setup.

### CODE REVIEW

**Comment by devinleighsmith** [comment]
Can you mark in the files what actually made the import start working again?

**Review by devinleighsmith — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -2,7 +2,7 @@
 # the repo. Unless a later match takes precedence,
 # @global-owner1 and @global-owner2 will be requested for
 # review when someone opens a pull request.
-* @devinleighsmith @Fosol @jdtoombs @goemen
+* @devinleighsmith @Fosol @jdtoombs @goemen @LawrenceLau2020
```
Comment (Fosol) [inline .github/CODEOWNERS:5]: Adding Lawrence so that he can start participating in code review

File: Makefile [hunk 2]
```diff
@@ -37,19 +37,23 @@ help:
 # Docker Development
 ##############################################################################
 
-docker: | build run ## Starts existing containers for local development
+restart: | stop build up ## Restart local docker environment
```
Comment (Fosol) [inline Makefile:40]: Cleaned up `make` file to be a little more organized and intuitive.

File: auth/README.md [hunk 3]
```diff
@@ -37,18 +36,17 @@ This will allow Keycloak to initialize with a new PostgreSQL database.
...
 
-> The default import during initialization no longer works. Start the keycloak container and manually import - [details here](./keycloak/README.md#Import%20Realm).
-
 ```conf
 # Keycloak configuration
 PROXY_ADDRESS_FORWARDING=true
 KEYCLOAK_USER={username}
 KEYCLOAK_PASSWORD={password}
-# KEYCLOAK_IMPORT=/tmp/realm-export.json -Dkeycloak.profile.feature.scripts=enabled -Dkeycloak.profile.feature.upload_scripts=enabled
+KEYCLOAK_IMPORT=/tmp/realm-export.json -Dkeycloak.profile.feature.scripts=enabled -Dkeycloak.profile.feature.upload_scripts=enabled
```
Comment (Fosol) [inline auth/README.md:44]: The import will now work, as long as the configuration file is valid.

File: auth/keycloak/README.md [hunk 4]
```diff
@@ -45,7 +39,7 @@ $ /opt/jboss/keycloak/bin/standalone.sh \
   -Dkeycloak.migration.action=export \
   -Dkeycloak.migration.provider=singleFile \
   -Dkeycloak.migration.realmName=pims \
-  -Dkeycloak.migration.file=/tmp/realm-export-new.json \
+  -Dkeycloak.migration.file=/tmp/realm-export.json \
```
Comment (Fosol) [inline auth/keycloak/README.md:42]: This must match the `docker-compose` volume name

File: auth/keycloak/README.md [hunk 5]
```diff
@@ -56,7 +50,7 @@ $ /opt/jboss/keycloak/bin/standalone.sh \
 Or (this doesn't appear to work on Windows)
 
 ```bash
-$ docker exec -it keycloak bash /opt/jboss/keycloak/bin/standalone.sh -Djboss.socket.binding.port-offset=100 -Dkeycloak.migration.action=export -Dkeycloak.migration.provider=singleFile -Dkeycloak.migration.strategy=OVERWRITE_EXISTING -Dkeycloak.migration.realmName=pims -Dkeycloak.migration.usersExportStrategy=REALM_FILE -Dkeycloak.migration.file=/tmp/realm-export-new.json
+$ docker exec -it keycloak bash /opt/jboss/keycloak/bin/standalone.sh -Djboss.socket.binding.port-offset=100 -Dkeycloak.migration.action=export -Dkeycloak.migration.provider=singleFile -Dkeycloak.migration.strategy=OVERWRITE_EXISTING -Dkeycloak.migration.realmName=pims -Dkeycloak.migration.usersExportStrategy=REALM_FILE -Dkeycloak.migration.file=/tmp/realm-export.json
```
Comment (Fosol) [inline auth/keycloak/README.md:53]: This must match the docker-compose volume name

File: docker-compose.yml [hunk 6]
```diff
@@ -2,23 +2,9 @@ version: "3.5"
 
 services:
   ####################### Keycloak #######################
-  keycloak-db:
```
Comment (Fosol) [inline docker-compose.yml:5]: Removing separate DB container for keycloak to simplify local setup.

File: frontend/public/keycloak.json [hunk 7]
```diff
@@ -1,6 +1,6 @@
 {
   "realm": "pims",
-  "auth-server-url": "http://keycloak:8080/auth",
+  "auth-server-url": "http://localhost:8080/auth",
```
Comment (Fosol) [inline frontend/public/keycloak.json:3]: Latest keycloak configuration no longer requires the same domain name for api and app

File: scripts/gen-env-files.sh [hunk 8]
```diff
@@ -18,14 +18,14 @@ if test -f "./auth/keycloak/.env"; then
 else
 echo \
 "PROXY_ADDRESS_FORWARDING=true
-DB_VENDOR=POSTGRES
-DB_ADDR=keycloak-db
-DB_DATABASE=keycloak
-DB_USER=$varKeycloakDb
-DB_PASSWORD=$passvar
+# DB_VENDOR=POSTGRES
```
Comment (Fosol) [inline scripts/gen-env-files.sh:21]: Separate DB is no longer required.

File: scripts/gen-env-files.sh [hunk 9]
```diff
@@ -18,14 +18,14 @@ if test -f "./auth/keycloak/.env"; then
...
+# DB_ADDR=keycloak-db
+# DB_DATABASE=keycloak
+# DB_USER=$varKeycloakDb
+# DB_PASSWORD=$passvar
 KEYCLOAK_USER=$varKeycloak
 KEYCLOAK_PASSWORD=$passvar
-# KEYCLOAK_IMPORT=/tmp/realm-export.json -Dkeycloak.profile.feature.scripts=enabled -Dkeycloak.profile.feature.upload_scripts=enabled
+KEYCLOAK_IMPORT=/tmp/realm-export.json -Dkeycloak.profile.feature.scripts=enabled -Dkeycloak.profile.feature.upload_scripts=enabled
```
Comment (Fosol) [inline scripts/gen-env-files.sh:28]: We want to import the config on startup now

File: tools/keycloak/sync/realm.development.json [hunk 10]
```diff
@@ -18,7 +18,7 @@
         "BearerOnly": true,
         "AuthorizationServicesEnabled": false,
         "RedirectUris": ["https://pims-dev.pathfinder.gov.bc.ca/api/*"],
-        "WebOrigins": ["+"],
+        "WebOrigins": ["*"],
```
Comment (Fosol) [inline tools/keycloak/sync/realm.development.json:21]: Allows us to use `localhost` instead of `keycloak`

File: tools/keycloak/sync/realm.json [hunk 11]
```diff
@@ -216,7 +216,7 @@
         "BearerOnly": true,
         "AuthorizationServicesEnabled": false,
         "RedirectUris": ["https://pims.gov.bc.ca/api/*"],
-        "WebOrigins": ["+"],
+        "WebOrigins": ["*"],
```
Comment (Fosol) [inline tools/keycloak/sync/realm.json:219]: Allows us to use `localhost` instead of `keycloak`

File: tools/keycloak/sync/realm.local.json [hunk 12]
```diff
@@ -18,7 +18,7 @@
         "BearerOnly": true,
         "AuthorizationServicesEnabled": false,
         "RedirectUris": ["http://localhost:5000/api/*"],
-        "WebOrigins": ["+"],
+        "WebOrigins": ["*"],
```
Comment (Fosol) [inline tools/keycloak/sync/realm.local.json:21]: Allows us to use `localhost` instead of `keycloak`

File: tools/keycloak/sync/realm.local.json [hunk 13]
```diff
@@ -40,7 +40,7 @@
         "RootUrl": "http://localhost:3000",
         "RedirectUris": ["http://localhost:3000/*"],
         "BaseUrl": "/",
-        "WebOrigins": ["http://localhost:3000"],
+        "WebOrigins": ["*"],
```
Comment (Fosol) [inline tools/keycloak/sync/realm.local.json:43]: Allows us to use `localhost` instead of `keycloak`

File: tools/keycloak/sync/realm.test.json [hunk 14]
```diff
@@ -18,7 +18,7 @@
         "BearerOnly": true,
         "AuthorizationServicesEnabled": false,
         "RedirectUris": ["https://pims-test.pathfinder.gov.bc.ca/api/*"],
-        "WebOrigins": ["+"],
+        "WebOrigins": ["*"],
```
Comment (Fosol) [inline tools/keycloak/sync/realm.test.json:21]: Allows us to use `localhost` instead of `keycloak`

**Comment by Fosol** [comment]
I've added comments to files.

**Review by jdtoombs — APPROVED** [review]

**Pull request merged**